tests/integration/path_planning.cpp

```diff
@@ -0,0 +1,338 @@
+
+    RRTPoint start = RRTPoint(state->config.getWaypoints()[0], 0);
+
+    int num_iterations = 500;
+    double search_radius = 9999;
+    double rewire_radius = 200;
+
+    auto s_time = std::chrono::high_resolution_clock::now();
+    // for (int i = 0; i < 2; i++) {
+    //     std::cout << "Iteration: " << i << std::endl;
+    //     // std::cout << "Start Running" << std::endl;
+    //     RRT rrt = RRT(start, goals, num_iterations, search_radius, rewire_radius,
+    //                   state->config.getFlightBoundary(), obstacles,
+    //                   OptimizationOptions{true, PATH_OPTIONS::RANDOM});
+
+    //     // print out stats
+    //     auto start_time = std::chrono::high_resolution_clock::now();
+    //     rrt.run();
+
+    //     // time the following function
+
+    //     auto end_time = std::chrono::high_resolution_clock::now();
+    //     std::chrono::duration<double> elapsed = end_time - start_time;
+    //     file << elapsed.count() << std::endl;
+    //     std::cout << "Time to run: " << elapsed.count() << "s" << std::endl;
+    //     std::vector<XYZCoord> path = rrt.getPointsToGoal();
+    //     std::cout << "Nodes: " << path.size() << std::endl;
+    // }
+    auto e_time = std::chrono::high_resolution_clock::now();
+    std::cout << "Total time: " << (e_time - s_time).count() << "s" << std::endl;
+
+    // start_time = std::chrono::high_resolution_clock::now();
+    // plot the path
+    auto start_time = std::chrono::high_resolution_clock::now();
+    RRT rrt = RRT(start, goals, num_iterations, search_radius, rewire_radius,
+                  state->config.getFlightBoundary(), obstacles,
+                  OptimizationOptions{true, PATH_OPTIONS::NEAREST});
+
+    // print out stats
+    std::cout << "num_iterations: " << num_iterations << std::endl;
+    std::cout << "search_radius: " << search_radius << std::endl;
+    std::cout << "rewire_radius: " << rewire_radius << std::endl;
+
+    rrt.run();
+
+    // time the following function3000
+
+    auto end_time = std::chrono::high_resolution_clock::now();
+    std::chrono::duration<double> elapsed = end_time - start_time;
+    file << elapsed.count() << std::endl;
+    std::chrono::duration<double> elapsed_total = e_time - s_time;
+
+    std::cout << "Time to run: " << elapsed.count() << "s" << std::endl;
+    std::cout << "End Running" << std::endl;
+
+    std::vector<XYZCoord> path = rrt.getPointsToGoal();
+    std::cout << "Path size: " << path.size() << std::endl;
+    std::cout << "Path length: " << (path.size() * POINT_SEPARATION) << std::endl;
+
+    std::cout << "Start Plotting" << std::endl;
+
+    // for (const XYZCoord& point : path) {
+    //     file << point.x << ", " << point.y << std::endl;
+    // }
+
+    PathingPlot plotter("pathing_output", state->config.getFlightBoundary(), obstacles[0], goals);
+
+    plotter.addFinalPolyline(path);
+    plotter.output("test_final_path", PathOutputType::BOTH);
```

Comment:
Jeez, the plotting takes forever. We should really figure this out to make debugging easier. Made an issue finally so we can look into it. https://github.com/tritonuas/obcpp/issues/128